daemon/cluster/executor/container/executor.go

```diff
@@ -136,23 +136,34 @@ func (e *executor) Describe(ctx context.Context) (*api.NodeDescription, error) {
 }
 
 func (e *executor) Configure(ctx context.Context, node *api.Node) error {
-	na := node.Attachment
-	if na == nil {
+
+	var ingressNA *api.NetworkAttachment
+	lbAttachments := make(map[string]string)
+
+	for _, na := range node.LbAttachments {
+		if na.Network.Spec.Ingress {
+			ingressNA = na
+		}
+		lbAttachments[na.Network.ID] = na.Addresses[0]
+	}
+
+	if ingressNA == nil {
 		e.backend.ReleaseIngress()
+		e.backend.ClearLBAttachments()
```

Comment:
is this function clearing the attachments of all the networks or only the Ingress?

Reply:
It is clearing all the attachments for all networks --- which is wrong.
I fixed this in the latest PR.  
The lb attachments are always "reset".  I.e. they are cleared and then updated with the latest node.lbattachments passed in from the swarm manager.